lib/tool_shed/webapp/model/__init__.py

```diff
@@ -201,14 +216,14 @@ def admin_role(self):
         raise Exception('Repository %s owned by %s is not associated with a required administrative role.' %
                         (str(self.name), str(self.user.username)))
 
-    def allow_push(self, app):
-        repo = hg_util.get_repo_for_repository(app, repository=self)
-        return repo.ui.config('web', 'allow_push')
+    @property
```

Comment:
I wouldn't use a property here since there is a `set_allow_push()` method which is not possible to decorate as a setter.